tempto-core/src/main/java/com/teradata/tempto/fulfillment/table/hive/tpcds/TpcdsDataSource.java

```diff
@@ -0,0 +1,145 @@
+ * limitations under the License.
+ */
+
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.teradata.tempto.fulfillment.table.hive.tpcds;
+
+import com.teradata.tempto.fulfillment.table.hive.HiveDataSource;
+import com.teradata.tempto.hadoop.hdfs.HdfsClient.RepeatableContentProducer;
+import com.teradata.tpcds.Results;
+import com.teradata.tpcds.Session;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static com.teradata.tpcds.Results.constructResults;
+import static java.lang.String.format;
+import static java.util.Collections.singleton;
+import static org.apache.commons.lang3.builder.EqualsBuilder.reflectionEquals;
+import static org.apache.commons.lang3.builder.HashCodeBuilder.reflectionHashCode;
+
+public class TpcdsDataSource
+        implements HiveDataSource
+{
+    private static final String DATA_REVISION = "v.1.0";
+
+    private final TpcdsTable table;
+    private final int scaleFactor;
+
+    public TpcdsDataSource(TpcdsTable table, int scaleFactor)
+    {
+        this.table = table;
+        this.scaleFactor = scaleFactor;
+    }
+
+    @Override
+    public String getPathSuffix()
+    {
+        return format("tpcds/sf-%d/%s", scaleFactor, table.name()).replaceAll("\\.", "_");
+    }
+
+    @Override
+    public Collection<RepeatableContentProducer> data()
+    {
+        return singleton(() -> new StringIteratorInputStream(generate()));
+    }
+
+    private Iterator<String> generate()
+    {
+        Session session = Session.getDefaultSession()
+                .withScale(scaleFactor)
+                .withParallelism(1)
+                .withTable(table.getTable())
+                .withNoSexism(false);
+        Results results = constructResults(table.getTable(), session);
+
+        return StreamSupport.stream(results.spliterator(), false)
+                .flatMap(rowBatch -> rowBatch.stream())
+                .map(row -> row.stream().collect(Collectors.joining("|")) + "|")
+                .flatMap(row -> Stream.of(row, "\n"))
+                .iterator();
+    }
+
+    @Override
+    public String revisionMarker()
+    {
+        return DATA_REVISION;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        return reflectionEquals(this, o);
```

Comment:
Cannot we simply use equals twice instead of relying on `reflectionEquals`? ;) Or if underlying `Table` does not have proper `equals` implemented it there instead of hacking it here?